meilidb-core/src/store/main.rs

```diff
@@ -114,6 +153,25 @@ impl Main {
         }
     }
 
+    pub fn put_fields_frequency(
+        self,
+        writer: &mut heed::RwTxn,
+        fields_frequency: &FreqsMap,
+    ) -> ZResult<()> {
+        self.main
+            .put::<Str, SerdeFreqsMap>(writer, FIELDS_FREQUENCY_KEY, fields_frequency)
+    }
+
+    pub fn fields_frequency(&self, reader: &heed::RoTxn) -> ZResult<Option<FreqsMap>> {
+        match self
+            .main
+            .get::<Str, SerdeFreqsMap>(&reader, FIELDS_FREQUENCY_KEY)?
```

Comment:
You do not need this ref, it is already a ref.
```suggestion
            .get::<Str, SerdeFreqsMap>(reader, FIELDS_FREQUENCY_KEY)?
```